skiplang/docs/SKStore.md

```diff
@@ -0,0 +1,142 @@
+# SKStore: A generic framework for reactive programming
+
+SKStore is a store made of collections called `directories`.
+Each directory contains entries associating values to a key.
+The keys can be of any type as long as they extend the class `SKStore.Key`, the values can be of any type as long as they extend the class `SKStore.File`.
+The APIs available for `SKStore` can be found in `Context.sk` and `Handle.sk`.
+
+## DirName
+
+Each directory (or collection) has a name, the name is defined with a slash separated path "/of/this/form/".
+The way to create a `DirName` is by using the primitive `DirName::create`.
+A directory name must start and finish with a slash, if it doesn't, an exception is thrown.
+The names are not used directly when manipulating directories, however, they come in handy when trying to debug the state that the system is in.
+One can navigate all the reactive values using familiar primitives (such as cd, ls etc ...).
```

Comment:
Can they? I've never seen that in the code and wasn't aware of any interactive shell-like thing

Reply:
I don't know, but will add a comment with your question.